File: scripts/run_import.py
#!/usr/bin/env python3

import os
import sys
import httpx

# Read SQL file
print("📖 Reading import file...")
with open('./data/grand_rapids_import_wrapped.sql', 'r') as f:
    sql_content = f.read()

print(f"📊 File size: {len(sql_content) / 1024:.2f} KB")

# Get Supabase URL from .env
supabase_url = None
with open('.env', 'r') as f:
    for line in f:
        if line.startswith('VITE_SUPABASE_URL='):
            supabase_url = line.strip().split('=', 1)[1]
            break

if not supabase_url:
    print("❌ Could not find VITE_SUPABASE_URL in .env")
    sys.exit(1)

print(f"🔗 Supabase URL: {supabase_url}")
print("⏳ Executing SQL via Supabase REST API (this may take 1-2 minutes)...\n")

# Note: This requires postgREST direct access or service role key
print("❌ Direct SQL execution requires service role key or PostgreSQL connection.")
print("\n💡 Alternative approaches:")
print("   1. Get service role key from Supabase dashboard")
print("   2. Use psql with database connection string")
print("   3. Use Supabase Studio SQL editor (copy/paste)")
print("\nFor now, please use the Supabase Studio SQL editor to run the import.")
print(f"📄 File location: ./data/grand_rapids_import_wrapped.sql")

File: scripts/run-import.mjs
#!/usr/bin/env node
/**
 * Run SQL Import Script
 * Executes SQL files via Supabase exec_sql RPC
 *
 * SECURITY REQUIREMENT: This script requires SUPABASE_SERVICE_ROLE_KEY
 * The exec_sql RPC function only accepts calls from service_role JWT claims.
 * Using anon key will result in permission denied errors.
 *
 * Usage:
 *   node scripts/run-import.mjs
 *   (requires SUPABASE_SERVICE_ROLE_KEY in .env)
 */

import { readFileSync } from "fs";
import { createClient } from "@supabase/supabase-js";
import dotenv from "dotenv";

// Load environment variables
dotenv.config();

const supabaseUrl = process.env.VITE_SUPABASE_URL;
// Support both env var names for backward compatibility
const supabaseServiceKey =
  process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_SERVICE_KEY;

if (!supabaseUrl) {
  console.error("❌ Missing VITE_SUPABASE_URL in .env file");
  process.exit(1);
}

if (!supabaseServiceKey) {
  console.error("❌ Missing SUPABASE_SERVICE_ROLE_KEY in .env file");
  console.error("💡 You need the service role key to execute raw SQL");
  console.error("   The exec_sql RPC requires service_role JWT claims.");
  console.error(
    "   Get it from: https://supabase.com/dashboard/project/aaqnanddcqvfiwhshndl/settings/api"
  );
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseServiceKey);

async function executeImport() {
  try {
    console.log("📖 Reading import file...");
    const sqlContent = readFileSync("./data/grand_rapids_import_wrapped.sql", "utf8");

    console.log(`📊 File size: ${(sqlContent.length / 1024).toFixed(2)} KB`);
    console.log("⏳ Executing SQL (this may take 1-2 minutes)...\n");

    // Execute the SQL using Supabase's REST API
    const { error } = await supabase.rpc("exec_sql", {
      sql_query: sqlContent,
    });

    if (error) {
      console.error("❌ Import failed:");
      console.error(JSON.stringify(error, null, 2));
      process.exit(1);
    }

    console.log("✅ Import completed successfully!");
    console.log("\n📋 Next steps:");
    console.log("   - Verify counts with validation queries");
    console.log(
      "   - Check that organizations, contacts, opportunities, and activities were created"
    );
  } catch (err) {
    console.error("❌ Unexpected error:", err.message);
    if (err.stack) console.error(err.stack);
    process.exit(1);
  }
}

executeImport();
